usecases/usecases.go

```diff
@@ -47,9 +47,10 @@ type UserInterface interface {
 	GetSnippetsByLanguage(language ProgrammingLanguage) ([]Snippet, error)
 	GetSnippet(snippet SnippetId) (Snippet, error)
 	DeleteSnippet(snippet SnippetId) error
-	Vote(snippet Snippet, vote int /* ±1 */) error
+	Vote(snippet SnippetId, vote int /* ±1 */) error
```

Comment:
Does voting happen anonymously? What could possible stop me from voting for a given snippet N times instead of just once?

Reply:
It's up to the implementer of this interface to return an error in this case. I don't know if this is the best decision though, will see later :)